### ISSUE TITLE
New C extension in Snow Leopard breaks C parsing

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ctypesgen/ctypesgen#6]
```
What steps will reproduce the problem?
1. Install Snow Leopard
2. Try to use ctypesgen against something that uses stdlib.h

What is the expected output? What do you see instead?
I should get no errors when creating the wrapper but instead I get a lot of 
Syntax errors.

The reason for this is as of Snow Leopard (OS X 10.6), there is a new C 
extension in the standard headers called blocks (http://bit.ly/8foNUD).  
Basically, this new extension allows you to create function pointers with a '^' 
character.  We need to either disable blocks functionality on OS X or add 
support for a function pointer using '^'.
```

Original issue reported on code.google.com by `jcscoob...@gmail.com` on 21 Jan 2010 at 4:48
